flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnOpFusion.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include "flashlight/fl/tensor/backend/jit/opt/backends/onednn/OneDnnOpFusion.h"
+
+#include "flashlight/fl/tensor/backend/jit/ir/BinaryNode.h"
+#include "flashlight/fl/tensor/backend/jit/ir/CustomNode.h"
+#include "flashlight/fl/tensor/backend/onednn/OneDnnBackend.h"
+#include "flashlight/fl/tensor/backend/onednn/OneDnnTensor.h"
+#include "flashlight/fl/tensor/backend/onednn/Utils.h"
+
+#include "dnnl.hpp"
+
+namespace fl {
+
+struct BinopInfo {
+  Node* rhsNode;
+  BinaryOp op;
+};
+
+struct OneDnnOpFusion::SearchState {
+  Node* searchRoot;
+  // Assume `searchRoot == binop2`
+  //
+  // x0  x1
+  //  \  /
+  //  binop1  x2
+  //     \  /
+  //    binop2
+  //
+  // accumulatedBinopInfos: { { binop2->op(), x2 }, { binop1->op(), x1 } }
+  std::vector<BinopInfo> accumulatedBinopInfos;
```

Comment:
Why not use a `std::array<2>` or the like if we know there are only going to be two children to binops?

I suppose each fusion pass will have its own independent abstractions on search; makes sense to me.

Reply:
> Why not use a std::array<2> or the like if we know there are only going to be two children to binops?

(If I understood correctly, you meant for `std::vector<BinopInfo>`? Because `BinopInfo` doesn't hold homogeneous fields.). The diagram shows only an example scenario. We could have arbitrary # of accumulated binops here (welp max 32 per OneDNN).

> will this structure eventually hold a vector<OpInfo> or something of the like?

That's what I have in mind, need to explore a bit more though. As explained in a comment above, ideally we need to support fuse heterogeneous ops in a single pass (to fully leverage OneDNN postops).